### PR TITLE
tikv: support dynamic modifications on `raftstore.store-io-pool-size`

### PR DESCRIPTION
**Pull request by LykxSassinator**
Signed-off-by: Lucasliang <nkcs_lykx@hotmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add docs on the dynamic modification supports to `raftstore.store-io-pool-size`.

Ref: https://github.com/tikv/tikv/issues/13964

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12765
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dynamic-config.md [hunk 1]
```diff
@@ -159,6 +159,7 @@ The following TiKV configuration items can be modified dynamically:
 | `raftstore.store-pool-size` | The number of threads in the pool that processes Raft, which is the size of the Raftstore thread pool |
 | `raftstore.apply-max-batch-size` | Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch. |
 | `raftstore.store-max-batch-size` | Raft state machines process requests for flushing logs into the disk in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can process the requests in one batch. |
+| `raftstore.store-io-pool-size` | The allowable number of threads that process Raft I/O tasks, which is the size of the StoreWriter thread pool (**NOT** modify this value from `> 0` to `== 0` or `== 0` to `> 0`). |
```
Comment (shichun-0415) [inline dynamic-config.md:162]:
```suggestion
| `raftstore.store-io-pool-size` | The allowable number of threads that process Raft I/O tasks, which is also the size of the StoreWriter thread pool (**DO NOT** modify this value from `> 0` to `== 0` or from `== 0` to `> 0`). |
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Comment by LykxSassinator** [comment]
@tonyxuqqi After https://github.com/tikv/tikv/pull/13965 merged, PTAL, thx.

File: dynamic-config.md [hunk 2]
```diff
@@ -159,6 +159,7 @@ The following TiKV configuration items can be modified dynamically:
 | `raftstore.store-pool-size` | The number of threads in the pool that processes Raft, which is the size of the Raftstore thread pool |
 | `raftstore.apply-max-batch-size` | Raft state machines process data write requests in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can execute the requests in one batch. |
 | `raftstore.store-max-batch-size` | Raft state machines process requests for flushing logs into the disk in batches by the BatchSystem. This configuration item specifies the maximum number of Raft state machines that can process the requests in one batch. |
+| `raftstore.store-io-pool-size` | The allowable number of threads that process Raft I/O tasks, which is also the size of the StoreWriter thread pool (**DO NOT** modify this value from `> 0` to `== 0` or from `== 0` to `> 0`) |
```
Comment (Oreoxmt) [inline dynamic-config.md:162]:
```suggestion
| `raftstore.store-io-pool-size` | The number of threads that process Raft I/O tasks, which is also the size of the StoreWriter thread pool (**DO NOT** modify this value from a non-zero value to 0 or from 0 to a non-zero value) |
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

**Comment by LykxSassinator** [comment]
@Oreoxmt cc, this PR can be merged, as https://github.com/tikv/tikv/pull/13965 has been merged.

**Pull request merged**